src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -249,6 +263,22 @@ export class ShellIntegrationAddon extends Disposable implements IShellIntegrati
 		return false;
 	}
 
+	private _doHandleITermSequence(data: string): boolean {
+		if (!this._terminal) {
+			return false;
+		}
+
+		const [command, value] = data.split(';');
+		switch (command) {
+			case ITermOscPt.SetMark: {
+				this._createOrGetCommandDetection(this._terminal).handleGenericCommand({ hoverMessage: value || '' });
+			}
```

Comment:
If we make any changes to the sequence like the hover message we should create our own, we have both ours and `ITermOscPt.SetMark` both call into `handleGenericCommand` though.

Reply:
hmm?

Reply:
maybe resolved w one of your comments below...?